folding/store.py

```diff
@@ -325,35 +237,32 @@ def upload_job(
 
 
 # Keep the Job and MockJob classes as they are, they work well with both implementations
-@dataclass
-class Job:
-    pdb: str
-    ff: str
-    box: str
-    water: str
-    hotkeys: list
-    job_type: str
-    created_at: pd.Timestamp
-    updated_at: pd.Timestamp
+class Job(BaseModel):
+    pdb_id: constr(min_length=4, max_length=10)
+    system_config: SystemConfig
+    s3_links: dict[str, str] | None
```

Comment:
I think it would be better to be: 
```suggestion
    s3_links: Union[Dict[str, str], None]
```